### PR TITLE
prototypic addition of TPC time series O2-4612

### PR DESCRIPTION
**Pull request by sawenzel**
https://its.cern.ch/jira/browse/O2-4612

### CODE REVIEW

File: MC/bin/o2dpg_sim_workflow.py [hunk 1]
```diff
@@ -1422,7 +1422,21 @@ def addQCPerTF(taskName, needs, readerCommand, configFilePath, objectsFile=''):
...
+
+   # TPC - time-series objects
+   # initial implementation taken from comments in https://its.cern.ch/jira/browse/O2-4612
+   # TODO: this needs to be made configurable (as a function of which detectors are actually present)
+   tpctsneeds = [ TPCRECOtask['name'],
+                  ITSTPCMATCHtask['name'],
+                  TOFTPCMATCHERtask['name'],
+                  PVFINDERtask['name']
+                ]
+   TPCTStask = createTask(name='tpctimeseries_'+str(tf), needs=tpctsneeds, tf=tf, cwd=timeframeworkdir, lab=["RECO"], mem='2000', cpu='1')
```
[the remark this answers is not in the document]
Comment (matthias-kleiner) [reply MC/bin/o2dpg_sim_workflow.py:1434]: Hello @sawenzel ,
the default is 4 threads, but you can also set it to 1 thread with the `--threads` option. I would assume the 2GB you assigned should be enough.

**Comment by miranov25** [comment]
Hello @sawenzel,

I'll take a look at the solution you've proposed and proceed with rebuilding O2.

Best regards,

Marian

**Comment by miranov25** [comment]
Hello @sawenzel,  @benedikt-voelkel and @matthias-kleiner 

Thank you for implementing the code. Unfortunately, when attempting to run the test as you suggested in JIRA, I encountered errors indicating that something might be missing.

* I executed your code as recommended:

```bash
#!/usr/bin/env bash
# Code suggested by Sandro and Peter
# MC->RECO->AOD for a simple pp min bias production

NWORKERS=${NWORKERS:-8}
MODULES="--skipModules ZDC"
SIMENGINE=${SIMENGINE:-TGeant4}

# Set up the MC workflow with this special Pythia8 config
# Create workflow with pp run conditions
${O2DPG_ROOT}/MC/bin/o2dpg_sim_workflow.py -eCM 14000 -col pp -gen pythia8 -proc cdiff -tf 2 \
                                           -ns 40 -e ${SIMENGINE} \
                                           -j ${NWORKERS} -interactionRate 500000 \
                                           -run 302000 -seed 624

# Run to TPC timeseries objects
${O2DPG_ROOT}/MC/bin/o2_dpg_workflow_runner.py -f workflow.json -tt tpctimes
```

* The error message I received is as follows:

```bash

[O2sim/latest-o2 O2DPG/latest-o2] /lustre/alice/users/miranov/NOTESData/alice-tpc-notes/JIRA/O2-4612 $> $NOTES/JIRA/O2-4612/mcTest.sh
Namespace(run=302000, productionTag='unknown', timestamp=-1, conditionDB='http://alice-ccdb.cern.ch', condition_not_after=3385078236000, orbitsPerTF=128, anchor_config='', dump_config='user_config.json', ns='40', gen='pythia8', proc='cdiff', trigger='', ini='', confKey='', interactionRate='500000', bcPatternFile='', meanVertexPerRunTxtFile='', eCM='14000', eA=-1, eB=-1, col='pp', field='ccdb', with_qed=False, ptHatMin=0, ptHatMax=-1, weightPow=-1, embedding=False, embeddPattern='@0:e1', nb=20, genBkg='', procBkg='heavy_ion', iniBkg='${O2DPG_ROOT}/MC/config/common/ini/basic.ini', confKeyBkg='', colBkg='PbPb', e='TGeant4', tf='2', production_offset=0, j=8, mod='--skipModules ZDC', with_ZDC=False, seed='624', o='workflow.json', noIPC=None, upload_bkg_to=None, use_bkg_from=None, early_tf_cleanup=False, pregenCollContext=False, no_combine_smaller_digi=False, no_combine_dpl_devices=False, no_mc_labels=False, no_tpc_digitchunking=False, with_strangeness_tracking=False, combine_tpc_clusterization=False, first_orbit=0, sor=-1, run_anchored=False, alternative_reco_software='', dpl_child_driver='', include_qc=False, include_local_qc=False, include_analysis=False, mft_reco_full=False, mft_assessment_full=False, fwdmatching_assessment_full=False, fwdmatching_4_param=False, fwdmatching_cut_4_param=False, fwdmatching_save_trainingdata=False)
** Using generic config **
INFO: Written additional config key parameters to JSON user_config.json
Diamond is
Using initialisation seed:  624
ALICEO2_CCDB_LOCALCACHE not set; setting to default /lustre/alice/users/miranov/NOTESData/alice-tpc-notes/JIRA/O2-4612/ccdb
Timeframe 1 seed:  130145275
Timeframe 2 seed:  130145276
=== There are 0 warnings ===
=== There are 0 errors ===
===> The workflow looks sane
Workflow saved at workflow.json
```

It appears that some tasks targeted for execution are missing from the workflow. Could you provide further assistance on how to resolve this issue? 

I checked also workflow.json, but this is long to paste to message. I did not see in that json anything about the time-series

@matthias-kleiner in case you can check I work at GSI lustre:
```
[O2sim/latest-o2 O2DPG/latest-o2] /lustre/alice/users/miranov/NOTESData/alice-tpc-notes/JIRA/O2-4612 $> less workflow.json
```

Best regards,
Marian

**Comment by miranov25** [comment]
@sawenzel 
I used your branch, I updated and recompiled full O2 - everything. 

I will start from clean /sw/

```
[O2sim/latest-o2 O2DPG/latest-o2] /aliceswO2/O2DPG $> git status .
On branch swenzel/O2-4612
Your branch is up to date with 'sawenzel/swenzel/O2-4612'.
```

**Comment by benedikt-voelkel** [comment]
There is no need to compile anything in case you are only developing `O2DPG`.
In fact, the simplest would be to
1. load `O2sim`
1. `export O2DPG_ROOT=/your/path/to/O2DPG` (here, that is really the path to where you cloned `O2DPG`)

**Comment by miranov25** [comment]
Hello @sawenzel and @benedikt-voelkel,

Thank you for your guidance.

Following the updates, recompilation, and rerun, I have now successfully generated both the time series and skimmed data. The cause of the issue over the weekend, which prevented the generation of the data, remains unclear to me.

The content of the example tree looks reasonable, as demonstrated by the example query on skimmed data.

I am interested in conducting a test production at GSI to compare the performance in MC with the performance maps in data. For Run2, we utilized a performance generator with Pythia Jets and a flat q/pt distribution+ random nuclei.

I have following questions:

* Would you recommend merging the current pull request now since the modification is functioning, and address further production-related changes in a separate pull request?
* I am inclined to run a small-scale production at GSI. If you do not have a specific "performance setup," should I simply run your test with an increased number of events?


image

**Comment by sawenzel** [comment]
I am merging it. @miranov25 : I think you could use ns=2000pp events per timeframe and tf=5 timeframes as reasonable (single slot) job size for test productions. Each job needs 8cores allocated.

**Pull request merged**